openapi_builder/parsers/docstring.py

```diff
@@ -0,0 +1,172 @@
+import ast
+import inspect
+import os
+from typing import Optional
+import sys
+
+
+class DocStringParser:
+    """
+    Parses a python file and returns the docstring.
+
+    This class can be instantiated from a given class, or from a given filename.
+    Usage:
+    >>> class A:
+    >>>    '''Some docstring for this class.'''
+    >>>
+    >>>    a = 42
+    >>>    '''Description of A.'''
+    >>> parser = DocStringParser.from_class(A)
+    >>> parser.parse()
+
+    Retrieving the results:
+    >>> parser.result
+    >>> {}
+
+    Limitations:
+    - The class must be placed in a file (doesn't work from a python shell)
+    - The class must be defined in the root content of a file (a class
+      definition in a function is not supported).
+    """
+
+    def __init__(self, module, prefix=None):
+        self.module = module
+        self.prefix = prefix
+        self.import_names = {}
+        """Dict to get the full import name:
+        i.e. 'import typing as t' results in: {'t': 'typing'}
+        """
+        self.result = {}
+
+    @classmethod
+    def from_class(cls, class_to_process, prefix=None):
```

Comment:
If you add `@functools.lru_cache(maxsize=64)` you can speed up the processing and reduce the printing on line 97.